plugins/easyimage/plugin.js

```diff
@@ -30,105 +63,170 @@
 			};
 		}
 
-		function createCommand( exec, refreshCheck, forceSelectionCheck ) {
+		function createButton( button ) {
+			editor.ui.addButton( button.name, {
+				label: button.label,
+				command: button.command,
+				toolbar: 'easyimage,' + ( button.order || 99 )
+			} );
+		}
+
+		function extractButtonInfo( style ) {
+			var info = {
+				label: style.label,
+				icon: style.icon,
+				iconHiDpi: style.iconHiDpi
+			};
+
+			delete style.label;
+			delete style.icon;
+			delete style.iconHiDpi;
+
+			return info;
+		}
+
+		function createCommand( options ) {
+			if ( options.button ) {
+				createButton( options.button );
+			}
+
 			return {
 				startDisabled: true,
 				contextSensitive: true,
 
 				exec: function( editor ) {
 					var widget = editor.widgets.focused;
 
-					exec( widget );
+					options.exec( widget );
 
-					if ( forceSelectionCheck ) {
+					if ( options.forceSelectionCheck ) {
 						// We have to manually force refresh commands as refresh seems to be executed prior to exec.
 						// Without this command states would be outdated.
 						editor.forceNextSelectionCheck();
 						editor.selectionChange( true );
 					}
 				},
 
-				refresh: createCommandRefresh( refreshCheck )
+				refresh: createCommandRefresh( options.refreshCheck )
 			};
 		}
 
-		editor.addCommand( 'easyimageFull', createCommand( function( widget ) {
-			widget.setData( 'type', 'full' );
-		}, function( widget ) {
-			return isFullImage( widget );
-		}, true ) );
-
-		editor.addCommand( 'easyimageSide', createCommand( function( widget ) {
-			widget.setData( 'type', 'side' );
-		}, function( widget ) {
-			return isSideImage( widget );
-		}, true ) );
-
-		editor.addCommand( 'easyimageAlt', new CKEDITOR.dialogCommand( 'easyimageAlt', {
-			startDisabled: true,
-			contextSensitive: true,
-			refresh: createCommandRefresh()
-		} ) );
-	}
+		function createStyleCommand( editor, name, styleDefinition ) {
+			var button = extractButtonInfo( styleDefinition ),
+				style;
 
-	function addToolbar( editor ) {
-		editor.ui.addButton( 'EasyimageFull', {
-			label: editor.lang.easyimage.commands.fullImage,
-			command: 'easyimageFull',
-			toolbar: 'easyimage,1'
-		} );
+			styleDefinition.type = 'widget';
+			styleDefinition.widget = 'easyimage';
+			styleDefinition.group = 'easyimage';
```

Comment:
What if we actually allowed to specify group? That would allow people to do these crazy border kind of things, without affecting the alignment.

So my thinking here is to make the `group` default to `'easyimage'` if not defined, but preserve `styleDefinition.group` if present.

That would be nice option for extensibility.

Reply:
With our current way of handling styles (buttons that allow only _applying_ styles) such change will result in non-reversible styles. I'd leave it the way it is for now.